### DISCUSSION TITLE
Refresh sObject Definition does not work for Custom Sharing Rule Objects

### DISCUSSION THREAD

**Discussion by monirkohi2006** [forcedotcom/salesforcedx-vscode#5789] [Ideas]
### Summary

Refresh sObject Definition does not work for Custom Sharing Rule Objects

### Steps To Reproduce:

1.  Define a custom sharing rule object. Ex: MyObject__Share shareRule = new MyObject__Share();
2. You should see the object and the instance with an underline stating "Variable does not exist: MyObject__Share".

### Expected result

SFDX should understand the share rule object as a valid object or should not underline it.

### Actual result

SFDX syntax checker doesn't understand the share rule object syntax.

### Additional information

![image](https://user-images.githubusercontent.com/32991357/92038424-33b20180-ed28-11ea-9548-e7eab44af759.png)

**VS Code Version**: Version: 1.48.2

**SFDX CLI Version**: sfdx-cli/7.68.6-d37008df83 darwin-x64 node-v12.18.3

**OS and version**: macOS Catalina & Version: 10.15.5

**Comment by khjuan** [comment]
I believe this is by design. See related (but different) issue i raised https://github.com/forcedotcom/salesforcedx-vscode/issues/2490.

Though i am not sure why it wanted to exclude all the __Share sobjects, I do see some legitimate uses of such sobjects in apex codes, especially the test classes. Perhaps this could be made a configurable feature of some sort?
